homeassistant/components/conversation/default_agent.py

```diff
@@ -223,22 +223,22 @@ async def async_process(self, user_input: ConversationInput) -> ConversationResu
         # Check if a trigger matched
         if isinstance(result, SentenceTriggerResult):
             # Gather callback responses in parallel
-            trigger_responses = await asyncio.gather(
-                *(
-                    self._trigger_sentences[trigger_id].callback(
-                        result.sentence, trigger_result
-                    )
-                    for trigger_id, trigger_result in result.matched_triggers.items()
+            trigger_callbacks = [
+                self._trigger_sentences[trigger_id].callback(
+                    result.sentence, trigger_result
                 )
-            )
+                for trigger_id, trigger_result in result.matched_triggers.items()
+            ]
 
             # Use last non-empty result as response.
             #
             # There may be multiple copies of a trigger running when editing in
             # the UI, so it's critical that we filter out empty responses here.
             response_text: str | None = None
-            for trigger_response in trigger_responses:
-                response_text = response_text or trigger_response
+            for trigger_future in asyncio.as_completed(trigger_callbacks):
```

Comment:
What happens with the other tasks when you break out of the for loop. Are they cancelled? Who is listening to their exceptions?

Reply:
I added more tests to confirm my expectations:

- `asyncio.as_completed` schedules all awaitables, and does not cancel them if you break out of the loop.
- The automation engine is listening to exceptions, so these will still be logged. But they won't bubble up to the agent here.